### PR TITLE
Allow FilterByCondition to condition also on subfields

### PR DESCRIPTION
**Pull request by dafnapension**
not only on keys in the instance

### CODE REVIEW

File: src/unitxt/operators.py [hunk 1]
```diff
@@ -1164,11 +1164,11 @@ def process(
...
-    Raises an error if a required key is missing.
+    Raises an error if a required key/subfield is missing.
 
     Args:
        values (Dict[str, Any]): Values that instances must match using the condition to be included in the output.
-       condition: the name of the desired condition operator between the key and the value in values ("gt", "ge", "lt", "le", "ne", "eq")
+       condition: the name of the desired condition operator between the key/subfield and the value in values ("gt", "ge", "lt", "le", "ne", "eq")
        error_on_filtered_all (bool, optional): If True, raises an error if all instances are filtered out. Defaults to True.
 
     Examples:
```
Comment (yoavkatz) [inline src/unitxt/operators.py:1174]: Add example of nested field

File: src/unitxt/operators.py [hunk 2]
```diff
@@ -1164,11 +1164,11 @@ def process(
 class FilterByCondition(SingleStreamOperator):
     """Filters a stream, yielding only instances for which the required values follows the required condition operator.
 
-    Raises an error if a required key is missing.
+    Raises an error if a required key/subfield is missing.
```
Comment (yoavkatz) [inline src/unitxt/operators.py:1167]:
```suggestion
    Raises an error if the field name is missing from the input instances.
```

File: src/unitxt/operators.py [hunk 3]
```diff
@@ -1164,11 +1164,11 @@ def process(
...
 
     Args:
        values (Dict[str, Any]): Values that instances must match using the condition to be included in the output.
-       condition: the name of the desired condition operator between the key and the value in values ("gt", "ge", "lt", "le", "ne", "eq")
+       condition: the name of the desired condition operator between the key/subfield and the value in values ("gt", "ge", "lt", "le", "ne", "eq")
```
Comment (yoavkatz) [inline src/unitxt/operators.py:1171]:
```suggestion
       condition: the name of the desired condition operator between the specified field  and the provided value ("gt", "ge", "lt", "le", "ne", "eq")
```

Comment (dafnapension) [reply src/unitxt/operators.py:1171]: Thanks for noticing! Corrected.

File: src/unitxt/operators.py (hunk 1 above)
Comment (dafnapension) [reply src/unitxt/operators.py:1174]: Done. Thanks.

File: src/unitxt/operators.py (hunk 2 above)
Comment (dafnapension) [reply src/unitxt/operators.py]: Done. thanks!

**Pull request merged**